doc/sphinx/source/developer_guide/abi.rst

```diff
@@ -0,0 +1,199 @@
+.. developer_guide abi
+
+=============
+ABI messaging
+=============
+
+Paparazzi use a custom publish/subscribe middleware to exchange data between software components named ``ABI``.
+
+.. note:: This middleware was named *ABI* for *AirBorne Ivy*
+
+General Idea
+------------
+
+* Give an easy way to allow software components to exchange data with minimum delay nor execution overhead.
+* The components don't need to know each other, they only need the message format.
+* Each subscriber set a callback function that will be called when new data are sent.
+
+.. warning:: This middleware is **NOT** thread safe. Threads are not widely used in Paparazzi but remember this if you use them.
+
+
+Message definition
+------------------
+
+The messages are described in ``conf/abi.xml`` analoguous to other messages in paparazzi. Name are unique, IDs starts from 0. "type" field should be a real C type.
+
+.. code-block:: xml
+
+    <class name="airborne">
+     <message name="DATA" id="0">
+      <field name="a" type="float"/>
+      <field name="b" type="struct bla"/>
+     </message>
+     ...
+    </class>
+
+Airborne code for subscriber
+----------------------------
+
+Include header and declare an ``abi_event`` as a global ``static`` variable (but **not** ``const``). Write the callback function with the proper prototype, matching the message.
+
+.. code-block:: C
+
+    #include "subsystems/abi.h"
+
+    static abi_event ev;
+
+    void data_cb(uint8_t sender_id, const float * a, const struct bla * b) {
```

Comment:
I checked in the code, there is a mistake here (and in the wiki as well). The fields are not necessarily pointers, unless typed like that in the XML file (this is a difference from pprzlink where all fields are turned to pointers). And they are not `const` either (all that was the original designed, but changed later).
So the callback should be:
```
void data_cb(uint8_t sender_id, float a, struct bla b)
```
Maybe it is worth adding a extra `c` field showing a pointer to something
```
<field name="c" type="uint32_t */>
void data_cb(uint8_t sender_id, float a, struct bla b, uint32_t * c)
```

Reply:
Rhaa, I noticed it this morning, I corrected it, but this one flew under the radar :smile: 
For the pointer example, I imagine users will see the examples of other messages using pointers, and in the wort case, the compiler error will be clear enough ?

Reply:
Well, you are right, an example is better if we you to have a good doc. This is done.